includes/class-wc-stripe-customer.php

```diff
@@ -393,6 +424,46 @@ public function get_sources() {
 		return empty( $sources ) ? [] : $sources;
 	}
 
+	/**
+	 * Gets saved payment methods for a customer using Intentions API.
+	 *
+	 * @param string $payment_method_type Stripe ID of payment method type
+	 *
+	 * @return array
+	 */
+	public function get_payment_methods( $payment_method_type ) {
+		if ( ! $this->get_id() ) {
+			return [];
+		}
+
+		$payment_methods = get_transient( self::PAYMENT_METHODS_TRANSIENT_KEY . $payment_method_type . $this->get_id() );
+
+		if ( false === $payment_methods ) {
+			$params   = WC_Stripe_UPE_Payment_Method_Sepa::STRIPE_ID === $payment_method_type ? '?expand[]=data.sepa_debit.generated_from.charge' : '';
+			$response = WC_Stripe_API::request(
+				[
+					'customer' => $this->get_id(),
+					'type'     => $payment_method_type,
+					'limit'    => 100, // Maximum allowed value.
```

Comment:
Should we open a ticket to paginate these? I think you mentioned the my account page is limited to showing only 10. Should we use 10 here? Is the limit imposed by Woo or something else?

Reply:
I believe this limit is imposed by Woo, so there isn't really anything we can do about it. You can see this when viewing saved payment methods from your accounts page: the maximum total tokens displayed is limited to ten, but if you have more tokens saved, they will start to appear as you delete the tokens that are shown. The limit also restricts us [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/2a8e109f6ab9aa6a04f4c87874ac458744ee2172/includes/class-wc-stripe-payment-tokens.php#L126), where the max total tokens returned inside the variable `$tokens` will also be ten, which is why I have to do [this](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/2a8e109f6ab9aa6a04f4c87874ac458744ee2172/includes/class-wc-stripe-payment-tokens.php#L223-L227) later on, to stop strange and bad things from happening.